test/commands/force/data/soql/query/dataSoqlQuery.nut.ts

```diff
@@ -48,25 +50,65 @@ function runQuery(query: string, options: QueryOptions = { json: true, ensureExi
 
 describe('data:soql:query command', () => {
   let testSession: TestSession;
+  let hubOrgUsername: string;
 
   before(async () => {
     testSession = await TestSession.create({
       setupCommands: [
         'sfdx force:org:create -f config/project-scratch-def.json --setdefaultusername --wait 10 --durationdays 1',
         'sfdx force:source:push',
+        'sfdx config:get defaultdevhubusername --json',
       ],
       project: { sourceDir: path.join('test', 'test-files', 'data-project') },
     });
     // Import data to the default org.
     execCmd(`force:data:tree:import --plan ${path.join('.', 'data', 'accounts-contacts-plan.json')}`, {
       ensureExitCode: 0,
     });
+
+    // get default devhub username
+    if (isArray<AnyJson>(testSession.setup)) {
+      hubOrgUsername = ensureString(
+        (testSession.setup[2] as { result: [{ key: string; value: string }] }).result.find(
+          (config) => config.key === 'defaultdevhubusername'
+        )?.value
+      );
+    }
   });
 
   after(async () => {
     await testSession?.clean();
   });
 
+  describe('data:soql:query respects maxQueryLimit config', () => {
+    it('should return 1 account record', () => {
+      // set maxQueryLimit to 1 globally
+      shell.exec('sfdx config:set maxQueryLimit=1 -g', { silent: true });
+
+      const result = runQuery('SELECT Id, Name, Phone FROM Account', { json: true }) as QueryResult;
+
+      expect(result.records.length).to.equal(1);
+      verifyRecordFields(result?.records[0], ['Id', 'Name', 'Phone', 'attributes']);
+    });
+
+    it('should return 3756 ScratchOrgInfo records', () => {
+      //
+      // set maxQueryLimit to 3756 globally
+      shell.exec('sfdx config:set maxQueryLimit=3756 -g', { silent: true });
+
+      const soqlQuery = 'SELECT Id FROM ScratchOrgInfo';
+      const queryCmd = `force:data:soql:query --query "${soqlQuery}" --json --targetusername ${hubOrgUsername}`;
```

Comment:
`runQuery` doesn't allow to pass a username, this is the only NUT using the hub so I just copy-pasted some checks in `runQuery` instead of refactor everything.